Sources/SpeziOnboarding/OnboardingConstraint.swift

```diff
@@ -0,0 +1,24 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import Spezi
+
+
+/// A Constraint which all `Standard` instances must conform to when using the Spezi Onboarding module.
+public protocol OnboardingConstraint: Standard {
+    /// Adds a new exported consent form represented as `Data` to the `Standard` conforming to ``OnboardingConstraint``.
+    /// 
+    /// - Parameter consent: The exported consent form represented as `Data` that should be added.
+    func store(consent: Data) async
+    
+    /// Loads the exported consent form represented as `Data` from the ``OnboardingDataSource``.
+    ///
+    /// - Returns: The loaded consent data.
+    func loadConsent() async throws -> Data
```

Comment:
I am wondering if this should be part of the full constraint. If I correctly interpret it here, it is more used for debugging and testing purposes so it might be implemented in the Standard of the example app but maybe not needed for every application that uses Spezi Onboarding?

Reply:
I thought about this design choice quite a bit as well. 
IMO, it makes sense to have both functions (so `store` and `load`) in the constraint as developers most probably want to also do something with the consent form later on, e.g. making it accessible in the application's user settings, ..
However, I don't have a strong opinion here, if you feel that this shouldn't be a part of the constraint, we can only opt for the `store` function.

Reply:
I think it would be great if developers implement this functionality but if we don't use it anywhere in the framework to e.g. provide a built-in view that displays the consent document or something I would not include it in the constraint. I would only ask developers to implement functionality that is necessary for the module to work and function. Suggestion them to write a load functionality in the docs sounds good to me. Or, alternatively, provide a great built-in functionality that uses this function to e.g. display a reusable view and then include it in the constraint as a requirement to provide the built-in functionality.